crates/ruff_db/src/system/path.rs

```diff
@@ -609,6 +609,22 @@ impl SystemVirtualPath {
         SystemVirtualPathBuf(self.0.to_string())
     }
 
+    /// Extracts the file extension, if possible.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use ruff_db::system::SystemVirtualPath;
+    ///
+    /// assert_eq!(None, SystemVirtualPath::new("Untitled-1").extension());
+    /// assert_eq!("ipynb", SystemVirtualPath::new("Untitled-1.ipynb").extension().unwrap());
+    /// ```
+    ///
+    /// See [`Path::extension`] for more details.
+    pub fn extension(&self) -> Option<&str> {
```

Comment:
I think the full string that VS code uses is `untitled://Untitled-1`. Can we add a test for this?

Reply:
I just checked in VS Code and it's
* `untitled:Untitled-1`
* `vscode-notebook-cell:Untitled-1.ipynb`

I'll update the test.

Reply:
Interesting, https://github.com/neovim/neovim/issues/21276 suggests that it should be a URL with the format `untitled://`

Reply:
I've added these test cases.